singer_sdk/streams/sql.py

```diff
@@ -825,9 +830,33 @@ def _adapt_column_type(
         Raises:
             NotImplementedError: if altering columns is not supported.
         """
-        current_type = self._get_column_type(full_table_name, column_name)
+        current_type: sqlalchemy.types.TypeEngine = self._get_column_type(
+            full_table_name, column_name
+        )
+
+        # Check if the existing column type and the sql type are the same
+        if isinstance(sql_type, type(current_type)):
+            # The current column and sql type are the same
+            # Nothing to do
+            return
+        # Check if the existing column generic type
+        # and sql generic type are the same
+        elif isinstance(sql_type.as_generic(), type(current_type.as_generic())):
```

Comment:
I'm not super clear on when `as_generic` would be useful. With something like NVARCHAR in postgres you get

```
(Pdb) abca=NVARCHAR(length=100)
(Pdb) abca
NVARCHAR(national=True, length=100, _warn_on_bytestring=True, _expect_unicode=True)
(Pdb) str(abca)
'NVARCHAR(100)'
(Pdb) abca.as_generic()
Unicode(length=100)
```

Not sure when this comes in handy for type checking? Maybe when comparing to json types, but in the unicode context it doesn't make a lot of sense

Reply:
I am using this to remove dialect specifics.  There were instances where the target `current_type` was `postgres` specific.  The `as_generic()` gives back the base TypeEngine the dialect specific version is based on.  The thought is if you have a `INT` column in `mysql` it will have the same generic class base as a `postgres`.  That is my very basic understanding of how it works at least.

Reply:
I understand and I'm not sure why that matters when we're only comparing data types from a single target with each target that's written. 🤷  I saw it in other places too I just don't understand why I guess there may be a great reason!

Reply:
I think I see why.  The `_get_column_type()` is based on `get_table_columns()' which gives back a dictionary that has a `column name` and the `sqlalchemy.Column` .  It then returns the a `TypeEngine` cast as that `sqlalchemy.Column` which would be dialect specific.  Then you need to use `as_generic()` to bring them back to the base TypeEngine to compare.